### DISCUSSION TITLE
Seeking Guidance: Error Encountered During libMesh Installation

### DISCUSSION THREAD

**Discussion by PengWei97** [libMesh/libmesh#3837] [General]
Dear libMesh Forum Members,

I hope this message finds you well. I am writing to seek guidance regarding an error encountered during the installation of libMesh.

When executing the command `make -j35` in the terminal during the libMesh installation process, I encountered the following error message:

```
In file included from /home/pw-moose/miniforge/envs/moose/x86_64-conda-linux-gnu/sysroot/usr/include/math.h:46,
                 from /home/pw-moose/miniforge/envs/moose/x86_64-conda-linux-gnu/include/c++/10.4.0/cmath:45,
                 from /home/pw-moose/miniforge/envs/moose/x86_64-conda-linux-gnu/include/c++/10.4.0/complex:44,
                 from ./include/libmesh/libmesh_common.h:53,
                 from ./include/libmesh/function_base.h:24,
                 from ./include/libmesh/parsed_function.h:22,
                 from ../src/apps/fparser_parse.C:23:
/usr/include/bits/mathdef.h:19:5: error: #error "Never use <bits/mathdef.h> directly; include <complex.h> instead"
```

I have attached the complete log file containing the terminal output for your reference. [bug_master.log](https://github.com/libMesh/libmesh/files/15062283/bug_master.log)

I would greatly appreciate any assistance or insights on how to resolve this issue and successfully complete the installation of libMesh. Your expertise and guidance on this matter would be immensely valuable to me.

Thank you very much for your time and assistance. 

Best regards,

Wei

**Comment by jwpeterson** [comment]
This error looks like it is coming from the math.h on your system, as `<bits/mathdef.h>` is not included by any files in the libmesh source.

**Reply by PengWei97** [reply]
Thank you very much for your prompt response and insights.

After carefully considering your suggestion, I suspect that the issue may indeed be related to the version discrepancy of gcc between my system and the moose environment. Specifically, I noticed that the GCC version installed via conda-forge (10.4.0) differs from the version installed via apt (11.4.0) on my Ubuntu system.

To further investigate this, I attempted to update my GCC version within the moose environment using the command `conda install -c conda-forge gcc=11.4.0`. However, the operation failed with the following message:

```
Could not solve for environment specs
The following packages are incompatible
├─ gcc 11.4.0**  is installable and it requires
│  └─ gcc_impl_linux-64 11.4.0.*  with the potential options
│     ├─ gcc_impl_linux-64 11.4.0 would require
│     │  └─ libsanitizer 11.4.0 h4dcbe23_0, which can be installed;
│     ├─ gcc_impl_linux-64 11.4.0 would require
│     │  └─ libsanitizer 11.4.0 h4dcbe23_1, which can be installed;
│     ├─ gcc_impl_linux-64 11.4.0 would require
│     │  └─ libsanitizer 11.4.0 h4dcbe23_2, which can be installed;
│     ├─ gcc_impl_linux-64 11.4.0 would require
│     │  └─ libsanitizer 11.4.0 h4dcbe23_3, which can be installed;
│     ├─ gcc_impl_linux-64 11.4.0 would require
│     │  └─ libsanitizer 11.4.0 h4dcbe23_4, which can be installed;
│     ├─ gcc_impl_linux-64 11.4.0 would require
│     │  └─ libsanitizer 11.4.0 h4dcbe23_5, which can be installed;
│     └─ gcc_impl_linux-64 11.4.0 would require
│        └─ libsanitizer 11.4.0 hc2b0fca_6, which can be installed;
└─ moose-dev is not installable because it requires
   ├─ moose-libmesh [2023.04.19 build_0|2023.04.19 build_1|2023.06.19 build_0|2023.06.23 build_0|2023.07.26 build_0], which requires
   │  └─ moose-mpich [4.0.2 build_7|4.0.2 build_8], which requires
   │     └─ libsanitizer 10.4.0 h5246dfb_19, which conflicts with any installable versions previously reported;
   └─ moose-peacock [2023.04.11.* |2023.08.17.* |...|2023.11.29.* ], which requires
      └─ moose-mpich [4.0.2 build_10|4.0.2 build_11|...|4.0.2 build_9], which cannot be installed (as previously explained)..
```

At this point, I'm unsure how to proceed further. Could you please provide any additional guidance or suggestions on how to resolve this issue?

Thank you once again for your assistance.

Best regards,
Wei

**Reply by roystgnr** [reply]
You're not going to be able to upgrade the conda-MOOSE dependencies in any way that conflicts with the conda-MOOSE (inside that particular conda environment, anyway).

You should report this on the MOOSE github page.  It's not at all a libMesh issue, and there's at least a better chance someone might be able to help you (and possibly be helped by you, if you're the first to notice an impending conflict) there.

**Reply by PengWei97** [reply]
OK, I will upload the same question to the moose forum now.

Thanks.
